### DISCUSSION TITLE
v3.1.2

### DISCUSSION THREAD

**Discussion by osy** [osy86/HaC-Mini#938] [Releases]
## Changes (v3.1.0)
* macOS Monterey support
* Fix warning message on OC boot (#585)
* SATA workaround is optional since it may break NVMe in some circumstances (#604)

## Changes (v3.1.1)
* Add Bluetooth support for Broadcom/Intel chips in Monterey (#923)

## Changes (v3.1.2)
* Fixed error during install of macOS 12.0 beta 8
* Fixed OTA updates of macOS 12.0 beta 6 and up
* Fixed Secure Boot

## Driver Updates (v3.1.0)
* AirportBrcmFixup 2.1.2 -> 2.1.3
* AppleALC 1.6.0 -> 1.6.4
* BrcmPatchRAM 2.5.8 -> 2.6.0
* IntelBluetoothFirmware 1.1.2 -> 2.0.0
* IntelMausi 1.0.6 -> 1.0.7
* Lilu 1.5.3 -> 1.5.6
* NVMeFix 1.0.7 -> 1.0.9
* OpenCore 0.6.9 -> 0.7.3
* OpenIntelWireless 1.3.0 -> 2.0.0
* Polaris22Fixup 1.3.3 -> 1.3.4
* VirtualSMC 1.2.3 -> 1.2.7
* WhateverGreen 1.4.9 -> 1.5.3

## Driver Update (v3.1.2)
* AppleALC 1.6.4 -> AppleALC 1.6.5
* IntelBluetoothFirmware 2.0.0 -> 2.0.1
* OpenCore 0.7.3 -> 0.7.4
* WhateverGreen 1.5.3 -> WhateverGreen 1.5.4

## Messages
* If you are upgrading from v2.17 and below, please read the notes at the top of [v3.0.1](https://github.com/osy/HaC-Mini/releases/tag/v3.0.1)
* [As announced a few months ago](https://github.com/osy/HaC-Mini/discussions/578), this project will be transitioning to a long-term support model (a few stability updates per year). Following Apple's own deprecation model, we expect the EOL of this project to be no earlier than 2023.

<hr /><em>This discussion was created from the release <a href='https://github.com/osy/HaC-Mini/releases/tag/v3.1.2'>v3.1.2</a>.</em>

**Comment by wangfuco** [comment]
I upgraded to 3.1.2 from 3.0.1, and after the reboot, my main nvme drive disappeared from the opencore boot options. The system is 10.15.7.

**Reply by x2358473** [reply]
I have the same problem.
Finally, boot with the old USB boot。

**Comment by Needham73** [comment]
Hi

After installing 3.1.2 Lilu started to crash during boot and now is on a reboot loop

**Reply by osy** [reply]
File a new issue with all the requested info.

**Reply by Needham73** [reply]
> File a new issue with all the requested info.

I forgot to say that it's with Monterey Beta. I believe that there is something that needs to be added to config.plist.
I restored config.plist from a backup and its working now.

**Reply by osy** [reply]
Still file an issue in case others have the same problem. What version of hac mini did you update from?

**Reply by Needham73** [reply]
> Still file an issue in case others have the same problem. What version of hac mini did you update from?

I updated from 3.1.1 but even then I had to do the patch because I updated WhateverGreen.kext and it also started to crash.
It has something to do with Lilu+WhateverGreen+AMD+Monterey.
I'm trying to find the instructions that I followed to solve it

**Reply by MokOopsing** [reply]
> > File a new issue with all the requested info.
> 
> I forgot to say that it's with Monterey Beta. I believe that there is something that needs to be added to config.plist. I restored config.plist from a backup and its working now.

could you plz shall your config.plist to me? I got the same issue

**Reply by Needham73** [reply]
> > > File a new issue with all the requested info.
> > 
> > 
> > I forgot to say that it's with Monterey Beta. I believe that there is something that needs to be added to config.plist. I restored config.plist from a backup and its working now.
> 
> could you plz shall your config.plist to me? I got the same issue

With Monterey Beta it was working ok but with the final release bluetooth stoped working, I'm still trying to figure it out.

**Reply by khanhdt123** [reply]
The Bluetooth problem has on the Beta 10 too.

**Reply by Needham73** [reply]
> The Bluetooth problem has on the Beta 10 too.

I jumped from Beta 8 to final release and it was working on beta 8. I tried the version 3.1.2 again but without bluetooth driver and it works (without bluetooth of course) so there must be something related to bluetooth

**Reply by Needham73** [reply]
> > The Bluetooth problem has on the Beta 10 too.
> 
> I jumped from Beta 8 to final release and it was working on beta 8. I tried the version 3.1.2 again but without bluetooth driver and it works (without bluetooth of course) so there must be something related to bluetooth

Solved it by updating BlueToolFixup.kext to 2.6.1 and placing it las in the kernel section of config.plist

**Reply by MokOopsing** [reply]
> > > The Bluetooth problem has on the Beta 10 too.
> > 
> > 
> > I jumped from Beta 8 to final release and it was working on beta 8. I tried the version 3.1.2 again but without bluetooth driver and it works (without bluetooth of course) so there must be something related to bluetooth
> 
> Solved it by updating BlueToolFixup.kext to 2.6.1 and placing it las in the kernel section of config.plist

Can you turn on Bluetooth after you turn it off?

**Reply by Needham73** [reply]
> > > > The Bluetooth problem has on the Beta 10 too.
> > > 
> > > 
> > > I jumped from Beta 8 to final release and it was working on beta 8. I tried the version 3.1.2 again but without bluetooth driver and it works (without bluetooth of course) so there must be something related to bluetooth
> > 
> > 
> > Solved it by updating BlueToolFixup.kext to 2.6.1 and placing it las in the kernel section of config.plist
> 
> Can you turn on Bluetooth after you turn it off?

If I manually turn it off I have to reboot to get it back on, other than that it's been working ok.

**Reply by MokOopsing** [reply]
> > > > > The Bluetooth problem has on the Beta 10 too.
> > > > 
> > > > 
> > > > I jumped from Beta 8 to final release and it was working on beta 8. I tried the version 3.1.2 again but without bluetooth driver and it works (without bluetooth of course) so there must be something related to bluetooth
> > > 
> > > 
> > > Solved it by updating BlueToolFixup.kext to 2.6.1 and placing it las in the kernel section of config.plist
> > 
> > 
> > Can you turn on Bluetooth after you turn it off?
> 
> If I manually turn it off I have to reboot to get it back on, other than that it's been working ok.

now you can try the latest beta build of BlueToolFixup.kext, it can fix this issue.

**Comment by x2358473** [comment]
@osy I'm sorry to bother you. The latest 3.1.2 does not display startup options on 10.15 systems. Could you please take a look at the problem? I like the 10.15 system very much and hope to use it for a long time.  If I'm bothering you. Please ignore it.
